bn-apps/memberships-management/membership-service/src/test/kotlin/com/r3/businessnetworks/membership/flows/AbstractFlowTest.kt

```diff
@@ -99,13 +99,16 @@ abstract class AbstractFlowTest(private val numberOfBusinessNetworks : Int,
         mockNetwork.runNetwork()
         return future.getOrThrow()
     }
-
-    fun runSelfIssueMembershipFlow(bnoNode : StartedMockNode, membershipMetadata : Any = SimpleMembershipMetadata(role = "DEFAULT")) : SignedTransaction {
-        val future = bnoNode.startFlow(SelfIssueMembershipFlow(membershipMetadata))
+  
+    fun runSelfIssueMembershipFlow(bnoNode : StartedMockNode) : SignedTransaction {
+        val membership = getMembership(bnoNode, bnoNode.identity(), bnoNode.identity())
```

Comment:
In this particular scenario the self issue flow should have failed, because membership for the BNO already exists on the ledger, and you are issuing a new one. It's good that you've added such test case :) Can you pls add a protection, so that no new membership can be issued if one already exists.
You can check how it is done here: https://github.com/corda/corda-solutions/blob/master/bn-apps/memberships-management/membership-service/src/main/kotlin/com/r3/businessnetworks/membership/flows/bno/RequestMembershipFlowResponder.kt#L37

Reply:
Protection has been added